code/eyetracking/eyetrackingrun.py

```diff
@@ -0,0 +1,961 @@
+        """
+        
+        if filename is None:
+            filename = f"sub-{self.participant:03d}_ses-{self.session:03d}_task-{self.task_name}_heatmap.pdf"
+
+        plt.rcParams["figure.figsize"] = [10, 6]
+        cmap = sns.color_palette("coolwarm", as_cmap=True)
+
+        if eye == "right":
+            filtered_samples = self.samples[
+                (self.samples["gx_right"] >= 0)
+                & (self.samples["gx_right"] <= self.screen_resolution[0])
+                & (self.samples["gy_right"] >= 0)
+                & (self.samples["gy_right"] <= self.screen_resolution[1])
+            ]
+
+            sns.kdeplot(
+                data=filtered_samples,
+                x="gx_right",
+                y="gy_right",
+                cmap=cmap,
+                fill=True,
+                cbar=True,
+                thresh=0,
+            )
+            plt.gca().invert_yaxis()
+            plt.xlabel("right eye x coordinate [pixels]")
+            plt.xlabel("right eye y coordinate [pixels]")
+        if notebook:
+            plt.show()
+        if save:
+            plt.savefig(os.path.join(path_save, filename))
+
+    def plot_delta(
+        self,
+        save: Optional[bool] = False,
+        path_save: Optional[str] = ".",
+        filename: Optional[str] = "blink_durations.pdf",
+        notebook: Optional[bool] = True,
+    ) -> None:
+        """
+        Plot the distribution of blink durations.
+
+        Parameters
+        ----------
+        save : bool, optional
+            Whether to save the plot as an image (default is False).
+        path_save : str, optional
+            Path to the directory where the image will be saved (default is current directory).
+        filename : str, optional
+            Name of the saved image file (default is "blink_durations.png").
+        notebook : bool, optional
+            If True, the plot will be displayed in a Jupyter notebook (default is True).
+
+        Returns
+        -------
+        None
+            Displays the plot or saves it.
+
+        Notes
+        -----
+        This method the blink occurences over time.
+        """
+        
+        blinks = self.events[self.events["blink"] == True]
+        blinks_start = blinks["start"].values
+        blinks_end = blinks["end"].values
+        timestamps = self.samples["time"].values
+
+        blinks_array = np.zeros_like(timestamps, dtype=int)
+        for start, end in zip(blinks_start, blinks_end):
+            blinks_array[(timestamps >= start) & (timestamps <= end)] = 1
+        plt.figure(figsize=(10, 6))
+        plt.plot(blinks_array)
+        plt.xlabel("Blink occurences over time")
+
+        if notebook:
+            plt.show()
+        if save:
+            output_filename = f"sub-{self.participant:03d}_ses-{self.session:03d}_task-{self.task_name}_{filename}"
+            plt.savefig(os.path.join(path_save, output_filename))
```

Comment:
Return the figure for the notebook to show and for further operations.

```suggestion
            plt.savefig(os.path.join(path_save, output_filename))

        return plt.gcf()
```